packagehandlers/packagehandlers_test.go

```diff
@@ -344,6 +344,43 @@ func TestUpdateDependency(t *testing.T) {
 				descriptorsToCheck: []string{"package.json"},
 			},
 		},
+
+		// Conan test cases
+		{
```

Comment:
all 3 of your test cases are really doing the same, you just check one of them, then the other, then them both, about the same fix.
Notice that when calling UpdateDependency it recognizes all existing descriptors and fix them all with the fixed version of openssl, therefore in your first testcase where you check only conanfile.txt - it fixes conanfile.py as well. In your second testcase it is exactly the same and you just check conanfile.py. The third testcase covers the two first testcases.
Overall:
If you wish to check a fix only in one of the files and NOT in the other you must put a unique dependency to fix in one file AND NOT IN THE OTHER.
If it is not important to you to check a fix in each file separately (and I cannot why you need to, they act by the same logic) just leave the third testcase only and delete the first two